staging/src/k8s.io/apiserver/pkg/apis/flowcontrol/bootstrap/default.go

```diff
@@ -327,7 +323,10 @@ var (
 			},
 		},
 	)
-	SuggestedFlowSchemaWorkloadLeaderElection = newFlowSchema(
+	// FIXME: I don't understand this rule at all - this overlaps with the one above.
```

Comment:
There is a difference in the target namespace (`kube-system` vs. every) in the first resourceRule.

In both FlowSchemas, the first resourceRule casts a pretty wide net, catching more than leader elections.  If I understand correctly, doing leader elections with Endpoints or ConfigMap objects is deprecated and not done by code in k/k.  If this is correct I would advocate removing the first resourceRule from both FlowSchemas.

Reply:
> If I understand correctly, doing leader elections with Endpoints or ConfigMap objects has been deprecated for a long time and not done by code in k/k. If this is correct I would advocate removing the first resourceRule from both FlowSchemas.

We migrated all core components out of it (there are some out-of-tree components still using it, but that should motivate them to switch). So yes - I agree we should remove this part of rule.

I would even claim that we should cherrypick it back, as currently, we still have things like endpoint-controller that could starve leader-election.

Also - the namespace for the second rule (leases) is NamespaceEvery for both rules (not sure if it's desired or not). So this is why I would remove the second rule.
Alternatively, we can change to kube-system in the first one, but I'm not sure if having two rules it buys us anything,

@MikeSpreitzer - if/when we agree on that - I will open a separate PR for it, as this is independent bugfix to ahve.

Reply:
So:
- the main reason why this was needed to past test was that kubemark was actually still using Endpoints instead of EndpointsSlices - I addressed that in https://github.com/kubernetes/kubernetes/pull/106740
- I already have a separate PR where we're discussing this change: https://github.com/kubernetes/kubernetes/pull/106725

But for now, I'm going to revert this change and double run tests against the current state - which I believe should actually pass now at head, by just enabling this mutating estimator.